packages/consumption/src/modules/attributes/local/attributeTypes/OwnIdentityAttribute.ts

```diff
@@ -0,0 +1,146 @@
+        if (!this.forwardedSharingDetails) return false;
+
+        const sharingDetailsWithPeer = this.forwardedSharingDetails.filter(
+            (sharingDetails) => sharingDetails.peer.equals(peer) && sharingDetails.deletionInfo?.deletionStatus !== EmittedAttributeDeletionStatus.DeletedByPeer
+        );
+        if (sharingDetailsWithPeer.length === 0) return false;
+
+        if (!excludeToBeDeleted) return true;
+
+        return sharingDetailsWithPeer.some((sharingDetails) => sharingDetails.deletionInfo?.deletionStatus !== EmittedAttributeDeletionStatus.ToBeDeletedByPeer);
+    }
+
+    public isDeletedOrToBeDeletedByForwardingPeer(peer: CoreAddress): boolean {
+        if (!this.forwardedSharingDetails) return false;
+
+        const sharingDetailsWithPeer = this.forwardedSharingDetails.filter((sharingDetails) => sharingDetails.peer.equals(peer));
+        if (sharingDetailsWithPeer.length === 0) return false;
+
+        const deletionStatuses = [EmittedAttributeDeletionStatus.DeletedByPeer, EmittedAttributeDeletionStatus.ToBeDeletedByPeer];
+
+        const hasSharingDetailsWithDeletionStatus = sharingDetailsWithPeer.some(
+            (sharingDetails) => sharingDetails.deletionInfo && deletionStatuses.includes(sharingDetails.deletionInfo.deletionStatus)
+        );
+        const hasSharingDetailsWithoutDeletionStatus = sharingDetailsWithPeer.some(
+            (sharingDetails) => !sharingDetails.deletionInfo || !deletionStatuses.includes(sharingDetails.deletionInfo.deletionStatus)
+        );
+        return hasSharingDetailsWithDeletionStatus && !hasSharingDetailsWithoutDeletionStatus;
+    }
+
+    public hasDeletionStatusUnequalDeletedByPeer(peer: CoreAddress): boolean {
+        if (!this.forwardedSharingDetails) return false;
+
+        const deletionStatuses = [
+            EmittedAttributeDeletionStatus.ToBeDeletedByPeer,
+            EmittedAttributeDeletionStatus.DeletionRequestSent,
+            EmittedAttributeDeletionStatus.DeletionRequestRejected
+        ];
+
+        return this.forwardedSharingDetails.some(
+            (sharingDetails) => sharingDetails.peer.equals(peer) && sharingDetails.deletionInfo && deletionStatuses.includes(sharingDetails.deletionInfo.deletionStatus)
+        );
+    }
+
+    public setForwardedDeletionInfo(deletionInfo: EmittedAttributeDeletionInfo | undefined, peer: CoreAddress, overrideDeleted = false): this {
+        const sharingDetailsForPeer = this.forwardedSharingDetails?.find(
+            (sharingDetails) =>
+                sharingDetails.peer.equals(peer) && (overrideDeleted || sharingDetails.deletionInfo?.deletionStatus !== EmittedAttributeDeletionStatus.DeletedByPeer)
+        );
+
+        if (!sharingDetailsForPeer) throw ConsumptionCoreErrors.attributes.cannotSetAttributeDeletionInfo(this.id, peer);
+
+        sharingDetailsForPeer.deletionInfo = deletionInfo;
+        return this;
+    }
+
+    public getForwardedPeers(includeToBeDeleted = false): CoreAddress[] {
```

Comment:
Should be more consistent with other naming.

Same for the OwnRelationshipAttribute and PeerRelationshipAttribute.

```suggestion
    public getForwardingPeers(includeToBeDeleted = false): CoreAddress[] {
```